@here/harp-mapview/lib/text/TextElementBuilder.ts

```diff
@@ -231,6 +233,7 @@ export class TextElementBuilder {
             tileOffset,
             offsetDirection
         );
+        textElement.dataSourceName = dataSourceName;
```

Comment:
pass it to the constructor to able to make it readonly.

Reply:
Done